### ISSUE TITLE
get access level of an entity wrong

### ISSUE THREAD

**Issue by videni** [oroinc/platform#860]
## Prepare  
Advertisement entity

A user has two roles ， ROLE_USER, ROLE_ORGANIAZATION

ROLE_ORGANIAZATION  for  advertisement entity has  GROUP_GLOBAL  permission
ROLE_USER for advertiesment entity has GROUP_NOE permission

## Use case
now let's us try to get `VIEW` access level of the advertisement for this user .

## Permission to mask map
```
 'MASK_VIEW_BASIC' =>
  int(1)
  'MASK_VIEW_LOCAL' =>
  int(2)
  'MASK_VIEW_DEEP' =>
  int(4)
  'MASK_VIEW_GLOBAL' =>
  int(8)
  'MASK_VIEW_SYSTEM' =>
  int(16)
  'GROUP_VIEW' =>
  int(31)
  'MASK_CREATE_BASIC' =>
  int(32)
  'MASK_CREATE_LOCAL' =>
  int(64)
  'MASK_CREATE_DEEP' =>
  int(128)
  'MASK_CREATE_GLOBAL' =>
  int(256)
  'MASK_CREATE_SYSTEM' =>
  int(512)
  'GROUP_CREATE' =>
  int(992)
  'MASK_EDIT_BASIC' =>
  int(1024)
  'MASK_EDIT_LOCAL' =>
  int(2048)
  'MASK_EDIT_DEEP' =>
  int(4096)
  'MASK_EDIT_GLOBAL' =>
  'GROUP_GLOBAL' =>
  int(8456)
 
```

## The cause
the ROLE_USER role will push the trigger mask to 16 , that is system level,  which is inferered from [PermissionGrantingStrategy.php#L294](https://github.com/oroinc/platform/blob/c1c08aaGROUP_GLOBALcb5b2cf380d8f9caaddb14ddb85fcf4c6/src/Oro/Bundle/SecurityBundle/Acl/Domain/PermissionGrantingStrategy.php#L294) .

   the ROLE_ORGNIZATION role  has GROUP_GLOBAL permission on the advertisement entity, it can't set the triggered make to 8 , because from this [code](https://github.com/oroinc/platform/blob/c1c08aacb5b2cf380d8f9caaddb14ddb85fcf4c6/src/Oro/Bundle/SecurityBundle/Acl/Domain/PermissionGrantingStrategy.php#L280), it can only set the triggered mask if  the access level of the required mask is bigger than previous,   so  we get a wrong access level. 

currently , I have to add following code, which fix this bug.
```
 if ($isGranting) {
    // override previous not granted triggered ace
    if (!$result) {
        $triggeredAce = $ace;
        $triggeredMask = $requiredMask;
    }
    $result = true;

    // the access is granted if there is at least one granting ACE
    if (null === $triggeredAce
        || $extension->getAccessLevel($requiredMask, null, $object)
        > $extension->getAccessLevel($triggeredMask, null, $object)
    ) {
        $triggeredAce = $ace;
        $triggeredMask = $requiredMask;
    }
}
```

I think this might be a bug.

**Comment by yurio** [comment]
Hello, @videni.
Thanks for reporting this issue.
We know about this problem and working on it.

**Comment by videni** [comment]
@yurio , hope it will be resolved soon.

**Comment by anyt** [comment]
The issue has been fixed in 2.6.30. Thank you
